### PR TITLE
eval_report this repo have evalution pipeline.

### PR DESCRIPTION
**Pull request by deepak-aryak9**
pipeline name - evaluate_slu
required params - {
"repo_name": "tvs-in",
"repo_branch": "sandbox", Not mandatory.
"test_dataset_path":"s3:"
}
with aliased file -- -
{
"repo_name": "tvs-in",
"repo_branch": "sandbox",
"test_dataset_path":"s3:/",
"alias_yaml_path": "alias.yaml"
}

Pipeline Testing -
Training -
https://kubeflow.skit.ai/_/pipeline/?ns=skit#/runs/details/d50c526f-114d-4a85-bc2c-8b779e91465d
Testing - 
https://kubeflow.skit.ai/_/pipeline/?ns=skit#/runs/details/405d12e1-b948-49e3-97c0-ecd5e85a1dc7
https://kubeflow.skit.ai/_/pipeline/?ns=skit#/runs/details/ed9730a7-cfe0-4f04-a448-829d5dbd4e28

### CODE REVIEW

**Comment by ltbringer** [comment]
Is it necessary to have the raw bucket path on the description?

- Buckets contain sensitive information.
- Bad actors crawl in search for leaking tokens, bucket paths etc.
- Even if a bucket is private, the path gives enough hints to discover other objects. Like one can guess there are prefixes same as employee names. Thanks to Linkedin I can loop through employees to discover paths that aren't private.

File: skit_pipelines/components/evaluate_slu_from_repo/__init__.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    import git
+    import pandas as pd
+    from loguru import logger
+
+    from skit_pipelines import constants as pipeline_constants
+    from skit_pipelines.components.download_repo import download_repo
+    from skit_pipelines.components.utils import (
+        pick_1st_tag, filter_dataset, alias_dataset, evaluate, create_dataset_path, execute_cli
+    )
+    from skit_pipelines.components.retrain_slu_from_repo.comparision_report_generator import \
```
Comment (harshithere) [inline skit_pipelines/components/evaluate_slu_from_repo/__init__.py:39]: Not needed imports

File: skit_pipelines/components/evaluate_slu_from_repo/__init__.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    filter_dataset(tagged_data_path, remove_intents)
+    alias_dataset(tagged_data_path, intent_alias_path)
+
+    # Testing
+    final_test_dataset_path = ""
+    if os.path.exists(tagged_data_path):
+        final_test_dataset_path = tagged_data_path
+
+
+    # testing the model that was just trained
```
Comment (harshithere) [inline skit_pipelines/components/evaluate_slu_from_repo/__init__.py:145]: Nitpick: Update the comments

File: skit_pipelines/components/evaluate_slu_from_repo/__init__.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    execute_cli(
+        f"cp {classification_report_path} {output_classification_report_path}"
+    )
+
+    execute_cli(
+         f"cp {confusion_matrix_path} {output_confusion_matrix_path}"
+    )
+
+    if os.path.exists(pipeline_constants.OLD_DATA):
```
Comment (harshithere) [inline skit_pipelines/components/evaluate_slu_from_repo/__init__.py:161]: This can be removed (you are not copying data to OLD_DATA path)

**Review by harshithere — CHANGES_REQUESTED** [review]
One major change is breaking the current retrain pipeline as well into components. Else, I'm assuming you have tested the changes and that both pipelines still work

File: skit_pipelines/components/evaluate_slu_from_repo/__init__.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    import git
+    from loguru import logger
+
+    from skit_pipelines import constants as pipeline_constants
+    from skit_pipelines.components.utils import (execute_cli)
+    from skit_pipelines.utils.normalize import comma_sep_str
+    from skit_pipelines.components.utils_slu import setup_repo, setup_project_config_repo, data_handler, testing
+
+    remove_intents = comma_sep_str(remove_intents)
+    compare = True if compare_branch else False
```
Comment (harshithere) [inline skit_pipelines/components/evaluate_slu_from_repo/__init__.py:39]: Nitpick: This compare variable is not needed. Values in `compare_branch` (checking if its !="") can do the same check

File: skit_pipelines/components/utils_slu.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+        project_config_local_path,
+        core_slu_repo_name,
+        repo_name
+    )
+
+    comparison_classification_report(classification_report_path, "", comparison_classification_report_path)
+    comparison_confusion_report(confusion_matrix_path, "", comparison_confusion_matrix_path)
+
+    if compare:
+        if os.path.exists("data.dvc"):
```
Comment (harshithere) [inline skit_pipelines/components/utils_slu.py:231]: Not needed condition. If `compare_branch` exisits, this value must be there

File: skit_pipelines/pipelines/evaluate_slu/__init__.py [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    tagged_job_data_op.execution_options.caching_strategy.max_cache_staleness = (
+        "P0D"  # disables caching
+    )
+
+    downloaded_alias_yaml_op = download_yaml_op(
+        git_host_name=pipeline_constants.GITLAB,
+        yaml_path=alias_yaml_path,
+    )
+
+    validate_evaluation_setup_op = evalution_slu_from_repo_op(
```
Comment (harshithere) [inline skit_pipelines/pipelines/evaluate_slu/__init__.py:159]: Question: How much time does evaluate take normally? If its a few minutes, then we dont need to validate the setup and can remove this

File: skit_pipelines/pipelines/evaluate_slu/__init__.py [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+        tagged_s3_data_op.outputs["output"],
+        tagged_job_data_op.outputs["output"],
+        downloaded_alias_yaml_op.outputs["output"],
+        bucket=BUCKET,
+        repo_name=repo_name,
+        compare_branch=compare_branch,
+        branch=repo_branch,
+        remove_intents=remove_intents,
+        labelstudio_project_ids=labelstudio_project_ids,
+        s3_paths=test_dataset_path,
```
Comment (harshithere) [inline skit_pipelines/pipelines/evaluate_slu/__init__.py:188]: You dont need `s3_path` and `labelstudio_project_ids` in evaluate_slu. Please remove these variables

**Review by harshithere — APPROVED** [review]
Good work.
Have a couple of minor changes. Can you attach some successful test runs in the description (both of new `evaluate_slu` and existing `retrain_slu`)

**Pull request merged**